metagraph/core/resolver.py

```diff
@@ -271,6 +284,30 @@ def _register_plugin_attributes_in_tree(
         concrete_types = set(
             concrete_types
         )  # copy; don't mutate original since we extend this set
+
+        self._register_wrappers_in_tree(tree, concrete_types, wrappers)
```

Comment:
The Resolver class is really long, and there seems to be two main purposes of its functions -- initial registry and runtime calling. To simplify, let's move all the registry logic to another class it ResolverRegistrar or something. All of its methods can be class methods. Or you can use regular instance methods and replace the class with a Singleton object after the fact. Either way, it will be in charge of registering the plugins with a resolver and all the messy logic can live there, allowing the Resolver to be focused on its usage by practitioners.

Reply:
https://github.com/metagraph-dev/metagraph/pull/169/commits/20cc843b36184b9e71e3b3d31aaaeca56db5bd56 

This commit creates a `ResolverRegistrar` class. All the methods are static methods.